### PR TITLE
Nosetests

### PR DESCRIPTION
**Pull request by anoopvalluthadam**
Changes that I have made
------------------------------------
@sylvestre 
1. I have added "templates/" to "tests/" folder
2. config_path parsing moved from GLOBAL to local
3. Few GLOBAL variables changed from global to local like;
        * email_nag.py: people in to main and passing it to methods
4. If we are doing a dry-run, I am using the default people dictionary from an option.dryrun, so that we 
   don't have to do to  any code level changes to make it run

### CODE REVIEW

File: auto_nag/scripts/phonebook.py [hunk 1]
```diff
@@ -35,11 +35,27 @@
 
 
 class PhonebookDirectory():
-
-    def __init__(self, config=CONFIG_JSON):
-        config = json.load(open(config, 'r'))
+    def __init__(self, TEST=False):
         print "Fetching people from phonebook..."
-        self.people = json.loads(requests.get(PEOPLE_URL, auth=(config['ldap_username'], config['ldap_password'])).content)
+        if TEST:
```
Comment (sylvestre) [inline auto_nag/scripts/phonebook.py]: I don't think that the test value should be hardcoded in the code itself.

**Comment by sylvestre** [comment]
You are mixing various changes in the same commits. I think the changes should be defined better. For example, a commit for the change in utils.py
A commit for the change in email_nag.py

File: auto_nag/scripts/phonebook.py [hunk 2]
```diff
@@ -11,9 +12,8 @@
 # In order to access the phonebook data
 
 
-MY_DIR = os.path.abspath(os.path.dirname(__file__))
-PEOPLE_FILENAME = os.path.join(MY_DIR, 'people.json')
-CONFIG_JSON = os.path.join(os.path.dirname(__file__), "configs/config.json")
+# MY_DIR = os.path.abspath(os.path.dirname(__file__))
```
Comment (sylvestre) [inline auto_nag/scripts/phonebook.py]: Why is that commented?

File: auto_nag/scripts/email_nag.py [hunk 3]
```diff
@@ -452,23 +462,23 @@ def add_to_managers(manager_email, query, info={}):
...
-                        msg_body += "\n=== %s ===\n" % k
-                    emailed_bugs.append(bug.id)
-                    msg_body += "http://bugzil.la/" + "%s -- assigned to: %s\n -- Last commented on: %s\n" % (bug.id, bug.assigned_to.real_name, bug.comments[-1].creation_time.replace(tzinfo=None))
-                msg = ("From: %s\r\n" % REPLY_TO_EMAIL
-                       + "To: %s\r\n" % REPLY_TO_EMAIL
-                       + "Subject: RelMan Attention Needed: %s\r\n" % options.email_subject
-                       + "\r\n"
-                       + msg_body)
-    sendMail(['release-mgmt@mozilla.com'], msg, options.mozilla_mail, options.email_password, options.dryrun)
+    if not options.roll_up:
```
Comment (sylvestre) [inline auto_nag/scripts/email_nag.py]: Please move that in a separate commit or PR

File: auto_nag/scripts/phonebook.py (hunk 2 above)
Comment (anoopvalluthadam) [inline auto_nag/scripts/phonebook.py]: I greped in our code base and we are not using it!

File: auto_nag/scripts/phonebook.py (hunk 1 above)
Comment (anoopvalluthadam) [inline auto_nag/scripts/phonebook.py]: Okay, I will get it as an argument.

File: auto_nag/scripts/phonebook.py (hunk 2 above)
Comment (sylvestre) [inline auto_nag/scripts/phonebook.py]: so, remove it :)

File: auto_nag/scripts/phonebook.py (hunk 1 above)
Comment (sylvestre) [inline auto_nag/scripts/phonebook.py]: yes, the sample manager/people data should be stored in a json file in the test directory.

File: auto_nag/scripts/phonebook.py (hunk 2 above)
Comment (anoopvalluthadam) [inline auto_nag/scripts/phonebook.py]: Sure!
